src/citrine/resources/delete.py

```diff
@@ -48,23 +57,45 @@ def _gemd_batch_delete(
         deleted.
 
     """
+    if len(id_list) > DELETE_SERVICE_MAX:  # we need to sort it
+        if any([not isinstance(x, BaseEntity) for x in id_list]):
+            raise TypeError(
+                "If more than {} deletes are requested, id_list must contain "
+                "only BaseEntities (objects & templates)".format(DELETE_SERVICE_MAX)
+            )
+        id_list = sorted(id_list, key=lambda x: writable_sort_order(x), reverse=True)
```

Comment:
I admit I am concerned about duplication of logic between here and the backend, but not sure we have much of a choice at present.